queries/javadoc/highlights.scm

```diff
@@ -0,0 +1,98 @@
+[
+ (tag_name)
+ "include"
+ "exclude"
+] @nospell @keyword
+
+(identifier) @nospell @variable
+
+(fragment) @nospell @variable.member
+
+(parameter
+  name: (identifier) @variable.parameter)
+
+(param_tag
+  parameter_name: (identifier) @variable.parameter)
+
+[
+ (boolean_type)
+ (integral_type)
+ (floating_point_type)
+] @nospell @type.builtin
+
+(module
+  (identifier) @module)
+
+(type
+  (identifier) @type)
+
+(type_parameter
+  (identifier) @type)
+
+(method
+  (identifier) @function)
+
+(((method
+    (identifier) @constructor)
+    (#match? @constructor "^[A-Z]")))
```

Comment:
Is this really necessary? We are trying to get away from convention-driven highlighting and prefer to rely on the actual grammar.

Also, `#match?` is never acceptable in Neovim for performance reasons; use `#lua-match?` with Lua patterns instead.

Reply:
no, not necessary. I was attempting to be consistent with parent language highilghts (java).

as far as the `#match?`, I can replace any that are left, after removing "conventions" matches. I only used them because they work with tree-sitter CLI.

Reply:
Yes, different editors behave differently, depending on the regex support. For Neovim, regex are particularly expensive since they cross the language barrier to Nvim core.